vgridjs/src/timeline_track.tsx

```diff
@@ -135,6 +135,80 @@ interface TimelineState {
   new_interval: Interval | null
 }
 
+interface TimelineNavigatorProps {
+  timeline_bounds: TimelineBounds,
+  timeline_width: number,
+  full_duration: number
+}
+
+@observer
+class TimelineNavigator extends React.Component<TimelineNavigatorProps, {}> {
+  private canvas_ref : React.RefObject<HTMLCanvasElement>;
+  private disposer : any;
+  
+  constructor (props: TimelineNavigatorProps) {
+    super(props);
+    this.canvas_ref = React.createRef();
+  }
+  
+  handleChange = (event: any) => {
+    if (event.target) {
+      let value  = event.target.valueAsNumber;
+      let max = parseInt(event.target.max);
+      let start = (this.props.full_duration * (value / this.props.timeline_width)) - (this.props.timeline_bounds.span()/2);
+      let end = (this.props.full_duration * (value / this.props.timeline_width)) + (this.props.timeline_bounds.span()/2);
+
+      if (start < 0) {
+        end -= start;
+        start = 0;
+      }
+      else if (end > this.props.full_duration) {
+        start -= end - this.props.full_duration;
+        end = this.props.full_duration
+      }
+      this.props.timeline_bounds.set_bounds(start, end);
+      this.forceUpdate();
+    }
+  }
+  
+  render_canvas() {
+    const canvas = this.canvas_ref.current;
+    if (canvas) {
+      canvas.style.display = 'block';
+      const ctx = canvas.getContext("2d");
+      if (ctx) {
+        ctx.fillStyle = "gray";
+        ctx.clearRect(0, 0, canvas.width, canvas.height);
+        let x = this.props.timeline_bounds.start * (canvas.width / this.props.full_duration);
+        let width = this.props.timeline_bounds.span() * (canvas.width / this.props.full_duration);
+        ctx.fillRect(x, 0, width, canvas.height);
+        ctx.strokeRect(0, 0, canvas.width, canvas.height);
+      }
+    }
+  }
+
+  componentDidMount() {
+    this.disposer = autorun(() => this.render_canvas());
+  }
+
+  componentDidUpdate() {
+    this.render_canvas();
+  }
+
+  componentWillUnmount() {
+    this.disposer();
+  }
+
+  render () {
+    return <div style={{width: this.props.timeline_width, height: 10, position: "relative"}}>
```

Comment:
Magic numbers (`height: 10`) should go into constants dictionary at top of the file